### PR TITLE
refactor(cli): use `cargo run` for `tauri dev`

### LINKED ISSUE #4724: [bug] `error while loading shared libraries: libonnxruntime.so.1.8.1: cannot open shared object file: No such file or directory`
### Describe the bug

`libonnxruntime.so.1.8.1` is downloaded and used by `onnxruntime`. In a normal rust project this works just fine, but I could not get it to work with tauri.

### Some info that might be relevant:
 - git repo for the dependency (`onnxruntime`) https://github.com/nbigaouette/onnxruntime-rs
 - `onnxruntime` depends on `onnxruntime-sys`
 - `onnxruntime-sys` has a custom build.rs that downloads `libonnxruntime.so.1.8.1`
 - everything works just fine in a new rust project, but this fails when using tauri
 
Minimal reproduction repo here: https://github.com/bogdibota/tauri-onnxruntime

### `dev` error message
```
dvk@pop-os:~/tmp/tauri-onnxruntime$ cargo tauri dev
        Info Watching /home/dvk/tmp/tauri-onnxruntime/src-tauri for changes...
   Compiling app v0.1.0 (/home/dvk/tmp/tauri-onnxruntime/src-tauri)
    Finished dev [unoptimized + debuginfo] target(s) in 22.06s
/home/dvk/tmp/tauri-onnxruntime/src-tauri/target/debug/tauri-onnxruntime: error while loading shared libraries: libonnxruntime.so.1.8.1: cannot open shared object file: No such file or directory
```

### `build -verbose` error message
```
dvk@pop-os:~/tmp/tauri-onnxruntime$ cargo tauri build --verbose
       Debug [globset] built glob set; 0 literals, 2 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] glob converted to regex: Glob { glob: "**/npm-debug.log*", re: "(?-u)^(?:/?|.*/)npm\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('n'), Literal('p'), Literal('m'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-debug.log*", re: "(?-u)^(?:/?|.*/)yarn\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-error.log*", re: "(?-u)^(?:/?|.*/)yarn\\-error\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('e'), Literal('r'), Literal('r'), Literal('o'), Literal('r'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] built glob set; 4 literals, 6 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 3 regexes
       Debug [globset] built glob set; 1 literals, 0 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] built glob set; 0 literals, 2 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] glob converted to regex: Glob { glob: "**/npm-debug.log*", re: "(?-u)^(?:/?|.*/)npm\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('n'), Literal('p'), Literal('m'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-debug.log*", re: "(?-u)^(?:/?|.*/)yarn\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-error.log*", re: "(?-u)^(?:/?|.*/)yarn\\-error\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('e'), Literal('r'), Literal('r'), Literal('o'), Literal('r'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] built glob set; 4 literals, 6 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 3 regexes
       Debug [globset] built glob set; 1 literals, 0 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] built glob set; 0 literals, 2 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] glob converted to regex: Glob { glob: "**/npm-debug.log*", re: "(?-u)^(?:/?|.*/)npm\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('n'), Literal('p'), Literal('m'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-debug.log*", re: "(?-u)^(?:/?|.*/)yarn\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-error.log*", re: "(?-u)^(?:/?|.*/)yarn\\-error\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('e'), Literal('r'), Literal('r'), Literal('o'), Literal('r'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] built glob set; 4 literals, 6 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 3 regexes
       Debug [globset] built glob set; 1 literals, 0 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] built glob set; 0 literals, 2 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] glob converted to regex: Glob { glob: "**/npm-debug.log*", re: "(?-u)^(?:/?|.*/)npm\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('n'), Literal('p'), Literal('m'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-debug.log*", re: "(?-u)^(?:/?|.*/)yarn\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-error.log*", re: "(?-u)^(?:/?|.*/)yarn\\-error\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('e'), Literal('r'), Literal('r'), Literal('o'), Literal('r'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] built glob set; 4 literals, 6 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 3 regexes
       Debug [globset] built glob set; 1 literals, 0 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] built glob set; 0 literals, 2 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] glob converted to regex: Glob { glob: "**/npm-debug.log*", re: "(?-u)^(?:/?|.*/)npm\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('n'), Literal('p'), Literal('m'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-debug.log*", re: "(?-u)^(?:/?|.*/)yarn\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-error.log*", re: "(?-u)^(?:/?|.*/)yarn\\-error\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('e'), Literal('r'), Literal('r'), Literal('o'), Literal('r'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] built glob set; 4 literals, 6 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 3 regexes
       Debug [globset] built glob set; 1 literals, 0 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] built glob set; 0 literals, 2 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] glob converted to regex: Glob { glob: "**/npm-debug.log*", re: "(?-u)^(?:/?|.*/)npm\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('n'), Literal('p'), Literal('m'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-debug.log*", re: "(?-u)^(?:/?|.*/)yarn\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-error.log*", re: "(?-u)^(?:/?|.*/)yarn\\-error\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('e'), Literal('r'), Literal('r'), Literal('o'), Literal('r'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] built glob set; 4 literals, 6 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 3 regexes
       Debug [globset] built glob set; 1 literals, 0 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
     Running [tauri_cli] Command `cargo  build --features custom-protocol --release`
   Compiling app v0.1.0 (/home/dvk/tmp/tauri-onnxruntime/src-tauri)
    Finished release [optimized] target(s) in 24.11s
       Debug [globset] built glob set; 0 literals, 2 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] glob converted to regex: Glob { glob: "**/npm-debug.log*", re: "(?-u)^(?:/?|.*/)npm\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('n'), Literal('p'), Literal('m'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-debug.log*", re: "(?-u)^(?:/?|.*/)yarn\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-error.log*", re: "(?-u)^(?:/?|.*/)yarn\\-error\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('e'), Literal('r'), Literal('r'), Literal('o'), Literal('r'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] built glob set; 4 literals, 6 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 3 regexes
       Debug [globset] built glob set; 1 literals, 0 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] built glob set; 0 literals, 2 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
       Debug [globset] glob converted to regex: Glob { glob: "**/npm-debug.log*", re: "(?-u)^(?:/?|.*/)npm\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('n'), Literal('p'), Literal('m'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-debug.log*", re: "(?-u)^(?:/?|.*/)yarn\\-debug\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('d'), Literal('e'), Literal('b'), Literal('u'), Literal('g'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] glob converted to regex: Glob { glob: "**/yarn-error.log*", re: "(?-u)^(?:/?|.*/)yarn\\-error\\.log[^/]*$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([RecursivePrefix, Literal('y'), Literal('a'), Literal('r'), Literal('n'), Literal('-'), Literal('e'), Literal('r'), Literal('r'), Literal('o'), Literal('r'), Literal('.'), Literal('l'), Literal('o'), Literal('g'), ZeroOrMore]) }
       Debug [globset] built glob set; 4 literals, 6 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 3 regexes
       Debug [globset] built glob set; 1 literals, 0 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
    Bundling [tauri_bundler::bundle::linux::debian] tauri-onnxruntime_0.1.0_amd64.deb (/home/dvk/tmp/tauri-onnxruntime/src-tauri/target/release/bundle/deb/tauri-onnxruntime_0.1.0_amd64.deb)
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("arch")], "arch")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("app_name")], "app_name")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("app_name")], "app_name")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("app_name")], "app_name")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("app_name")], "app_name")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("icon_path")], "icon_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("icon_path")], "icon_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("app_name")], "app_name")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("app_name")], "app_name")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("app_name")], "app_name")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("appimage_filename")], "appimage_filename")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("tauri_tools_path")], "tauri_tools_path")))
       Debug [handlebars::render] Rendering value: Path(Relative(([Named("app_name")], "app_name")))
    Bundling [tauri_bundler::bundle::linux::appimage] tauri-onnxruntime_0.1.0_amd64.AppImage (/home/dvk/tmp/tauri-onnxruntime/src-tauri/target/release/bundle/appimage/tauri-onnxruntime_0.1.0_amd64.AppImage)
     Running [tauri_bundler::bundle::common] Command `/home/dvk/tmp/tauri-onnxruntime/src-tauri/target/release/bundle/appimage/build_appimage.sh `
+ export ARCH=x86_64
+ ARCH=x86_64
+ APPIMAGE_BUNDLE_XDG_OPEN=0
+ APPIMAGE_BUNDLE_GSTREAMER=0
+ TRAY_LIBRARY_PATH=0
+ '[' x86_64 == i686 ']'
+ linuxdeploy_arch=x86_64
+ mkdir -p tauri-onnxruntime.AppDir
+ cp -r ../appimage_deb/data/usr tauri-onnxruntime.AppDir
+ cd tauri-onnxruntime.AppDir
+ mkdir -p usr/bin
+ mkdir -p usr/lib
+ [[ 0 != \0 ]]
+ [[ 0 != \0 ]]
++ dirname '{}'
+ find /usr/lib /usr/lib32 /usr/lib64 /usr/libexec /usr/libx32 -name WebKitNetworkProcess -exec mkdir -p . ';' -exec cp --parents '{}' . ';'
++ dirname '{}'
+ find /usr/lib /usr/lib32 /usr/lib64 /usr/libexec /usr/libx32 -name WebKitWebProcess -exec mkdir -p . ';' -exec cp --parents '{}' . ';'
++ dirname '{}'
+ find /usr/lib /usr/lib32 /usr/lib64 /usr/libexec /usr/libx32 -name libwebkit2gtkinjectedbundle.so -exec mkdir -p . ';' -exec cp --parents '{}' . ';'
+ wget -q -4 -N -O /home/dvk/.cache/tauri/AppRun https://github.com/AppImage/AppImageKit/releases/download/continuous/AppRun-x86_64
+ chmod +x /home/dvk/.cache/tauri/AppRun
+ cp /home/dvk/.cache/tauri/AppRun .
+ cp usr/share/icons/hicolor/256x256@2/apps/tauri-onnxruntime.png .DirIcon
+ ln -s usr/share/icons/hicolor/256x256@2/apps/tauri-onnxruntime.png tauri-onnxruntime.png
+ ln -s usr/share/applications/tauri-onnxruntime.desktop tauri-onnxruntime.desktop
+ cd ..
+ [[ 0 != \0 ]]
+ gst_plugin=
+ wget -q -4 -N -O /home/dvk/.cache/tauri/linuxdeploy-plugin-gtk.sh https://raw.githubusercontent.com/tauri-apps/linuxdeploy-plugin-gtk/master/linuxdeploy-plugin-gtk.sh
+ wget -q -4 -N -O /home/dvk/.cache/tauri/linuxdeploy-x86_64.AppImage https://github.com/tauri-apps/binary-releases/releases/download/linuxdeploy/linuxdeploy-x86_64.AppImage
+ chmod +x /home/dvk/.cache/tauri/linuxdeploy-plugin-gtk.sh
+ chmod +x /home/dvk/.cache/tauri/linuxdeploy-x86_64.AppImage
+ dd if=/dev/zero bs=1 count=3 seek=8 conv=notrunc of=/home/dvk/.cache/tauri/linuxdeploy-x86_64.AppImage
3+0 records in
3+0 records out
3 bytes copied, 0.000144287 s, 20.8 kB/s
+ OUTPUT=tauri-onnxruntime_0.1.0_amd64.AppImage
+ /home/dvk/.cache/tauri/linuxdeploy-x86_64.AppImage --appimage-extract-and-run --appdir tauri-onnxruntime.AppDir --plugin gtk --output appimage
linuxdeploy version 1-alpha (git commit ID 56760df), GitHub actions build 90 built on 2022-07-12 10:39:09 UTC

-- Creating basic AppDir structure -- 
Creating directory tauri-onnxruntime.AppDir/usr/bin/ 
Creating directory tauri-onnxruntime.AppDir/usr/lib/ 
Creating directory tauri-onnxruntime.AppDir/usr/share/applications/ 
Creating directory tauri-onnxruntime.AppDir/usr/share/icons/hicolor/ 
Creating directory tauri-onnxruntime.AppDir/usr/share/icons/hicolor/16x16/apps/ 
Creating directory tauri-onnxruntime.AppDir/usr/share/icons/hicolor/32x32/apps/ 
Creating directory tauri-onnxruntime.AppDir/usr/share/icons/hicolor/64x64/apps/ 
Creating directory tauri-onnxruntime.AppDir/usr/share/icons/hicolor/128x128/apps/ 
Creating directory tauri-onnxruntime.AppDir/usr/share/icons/hicolor/256x256/apps/ 
Creating directory tauri-onnxruntime.AppDir/usr/share/icons/hicolor/scalable/apps/ 

-- Deploying dependencies for existing files in AppDir -- 
Deploying dependencies for ELF file tauri-onnxruntime.AppDir/usr/bin/tauri-onnxruntime 
ERROR: Could not find dependency: libonnxruntime.so.1.8.1 
ERROR: Failed to deploy dependencies for existing files 
       Error [tauri_cli] failed to bundle project: error running appimage.sh: error running appimage.sh: `failed to run /home/dvk/tmp/tauri-onnxruntime/src-tauri/target/release/bundle/appimage/build_appimage.sh`
```

### Manual setup
1. `yarn create tauri-app`
2. add `onnxruntime = "0.0.14"` as dependency
3. use `onnxruntime` in `main.rs`
4. `yarn tauri dev`

The dev build will finish successfully, but right after it fails with: `error while loading shared libraries: libonnxruntime.so.1.8.1: cannot open shared object file: No such file or directory`

### Expected behavior

Basically not fail to build / start. I expect  `onnxruntime` to work with tauri.

### Platform and versions

```shell
dvk@pop-os:~/tmp/tauri-onnxruntime$ cargo tauri info

Environment
  › OS: Pop!_OS 22.04 X64
  › Node.js: 16.14.0
  › npm: 8.3.1
  › pnpm: Not installed!
  › yarn: 1.22.17
  › rustup: 1.24.3
  › rustc: 1.62.0
  › cargo: 1.62.0
  › Rust toolchain: stable-x86_64-unknown-linux-gnu 

Packages
  › @tauri-apps/cli [NPM]: 1.0.4
  › @tauri-apps/api [NPM]: 1.0.2
  › tauri [RUST]: 1.0.4,
  › tauri-build [RUST]: 1.0.4,
  › tao [RUST]: 0.12.1,
  › wry [RUST]: 0.19.0,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: ../dist

App directory structure
  ├─ .git
  ├─ dist
  └─ src-tauri
```
```


### Stack trace

_No response_

### Additional context

The library `onnxruntime` builds and works just fine in a simple rust project, on the same system.

### LINKED ISSUE #8169: [bug] cargo tauri dev command doesn't respect .cargo/config.toml [env] section settings
### Describe the bug

If you set environment variable through cargo `config.toml` file, like below, `cargo tauri dev` doesn't add it to the target app.

**.cargo/config.toml**
```toml
[env]
FOO = "bar"
```

### Reproduction

1. preprare barebone  tauri-app
```shell
cargo install create-tauri-app --locked
cargo create-tauri-app
```

2. edit  **tauri-app/src-tauri/src/main.rs**
```rust
#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]

#[tauri::command]
fn greet(name: &str) -> String {
    format!("Hello, {}! You've been greeted from Rust!", name)
}

fn main() {

    match std::env::var("FOO") {
        Ok(v) => println!("FOO={v}"),
        Err(e) => println!("Error: {:?}",e),
    }

    tauri::Builder::default()
        .invoke_handler(tauri::generate_handler![greet])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

3. edit **.cargo/config.toml**
```toml
[env]
FOO = "bar"
```

4. run `cargo tauri dev`

> /r/tauri-app> cargo tauri dev
        Info Watching tauri-app/src-tauri for changes...
   Compiling tauri-app v0.0.0 (tauri-app/src-tauri)
    Finished dev [unoptimized + debuginfo] target(s) in 2.55s
Error: NotPresent

### Expected behavior

Running command `cargo tauri dev`  should print below.

> /r/tauri-app> cargo tauri dev
        Info Watching tauri-app/src-tauri for changes...
   Compiling tauri-app v0.0.0 (tauri-app/src-tauri)
    Finished dev [unoptimized + debuginfo] target(s) in 2.55s
FOO=bar

### Platform and versions

```text
[✔] Environment
    - OS: Mac OS 10.14.5 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.70.0 (90c541806 2023-05-31)
    ✔ cargo: 1.70.0 (ec8a8a0ca 2023-04-25)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-apple-darwin (environment override by RUSTUP_TOOLCHAIN)
    - node: 20.5.0
    - yarn: 1.22.11
    - npm: 9.8.0

[-] Packages
    - tauri [RUST]: 1.5.2
    - tauri-build [RUST]: 1.5.0
    - wry [RUST]: 0.24.4
    - tao [RUST]: 0.16.5
    - tauri-cli [RUST]: 1.5.6
    - @tauri-apps/api : not installed!
    - @tauri-apps/cli [NPM]: 1.5.6

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../src
    - devPath: ../src
```

### Stack trace

_No response_

### Additional context

If we run tauri-app by running command `cd src-tauri; cargo run`, the app correctly prints the environment variable set by `.cargo/config.toml`.

### PR DESCRIPTION
**Pull request by FabianLars**
closes #4724
edit: closes #8169

Soooo i have small problem here. I debugged this for so long (hours) that i don't even remember why i was debugging it. The only difference between the current behavior and this PR is a new error message on ctrlc but i think this `cargo run` is printing that.

i would appreciate some testing and telling me what i missed.

p.s. the cargo check workflow will fail. didn't want to remove DevCleanup and manually_killed until someone tells me what i missed.

p.p.s. i'll try to do some refactoring after this gets merged. the whole cli dev/build impl is a bit convoluted 😅 

p.p.p.s idk why github messes up the diff this much, it's better in vscode for me

### CODE REVIEW

**Comment by FabianLars** [comment]
also, most of the changes were done last week, postponed due to system issues which is probably why i don't know what was wrong with it anymore x)

**Comment by FabianLars** [comment]
I left the DevProcess stuff as a todo because it requires touching code shared by mobile and desktop. I'd prefer to wait and see if we need anything of the "disabled" code after some real world testing and remove it later incl a small refactor rather than removing it now and reverting some stuff.

If you disagree i can do it still but idk when i have the time for that.

p.s. the lint workflow fails in unrelated parts of the codebase

**Review by lucasfernog — DISMISSED** [review]

**Comment by FabianLars** [comment]
(accidentally dismissed your review and can't merge my own prs in this repo without an approval)

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**
